### PR TITLE
Added PrepareUserInfoToTransmit callback to ReHLDS API

### PR DESCRIPTION
**Pull request by WPMGPRoSToTeMa**

### CODE REVIEW

File: rehlds/engine/sv_main.cpp [hunk 1]
```diff
@@ -3713,6 +3713,8 @@ void SV_FullClientUpdate(client_t *cl, sizebuf_t *sb)
 	info[sizeof(info) - 1] = 0;
 	Info_RemovePrefixedKeys(info, '_');
 
+	g_RehldsHookchains.m_PrepareUserInfoToTransmit.callChain(nullptr, GetRehldsApiClient(cl), info);
```
Comment (dreamstalker) [inline rehlds/engine/sv_main.cpp]: I think you should also pass the length of the info buffer to this chain, in order to manipulate that buffer safely.

**Comment by theAsmodai** [comment]
I think would be useful add to hookchain 'receiver' argument for customizations. For example, that different players can see different models. I once did this for CSDM FFA.

**Comment by WPMGPRoSToTeMa** [comment]
@theAsmodai then we need change following code to for loop:
```C++
{
	client->sendinfo = FALSE;
	client->sendinfo_time = realtime + 1.0;
	SV_ExtractFromUserinfo(client);
	SV_FullClientUpdate(client, &g_psv.reliable_datagram);
}
```
But what macro we need choose for code separation?

**Comment by theAsmodai** [comment]
It's a little more complicated than I thought. The recipient can be determined by msg and host_client, but for customization we must also change the destination buffer. Maybe will be better to make a separate hookchain for the whole SV_FullClientUpdate function, and implement customization logic in metamod module.

**Comment by theAsmodai** [comment]
@WPMGPRoSToTeMa, I created separate PR for it: https://github.com/dreamstalker/rehlds/pull/62

**Comment by dreamstalker** [comment]
Obsoleted by #62

**Pull request closed without merging**